kiliautoml/models/_hugging_face_text_classification_model.py

```diff
@@ -0,0 +1,236 @@
+            predictions_probability=proba_assets,
+        )
+
+        return job_predictions
+
+    def _train(
+        self,
+        assets: List[Dict],
+        job: Dict,
+        job_name: str,
+        model_framework: str,
+        model_name: str,
+        path: str,
+        clear_dataset_cache: bool,
+    ) -> float:
+
+        kili_print(job_name)
+        path_dataset = os.path.join(path, "dataset", "data.json")
+        kili_print(f"Downloading data to {path_dataset}")
+        if os.path.exists(path_dataset) and clear_dataset_cache:
+            os.remove(path_dataset)
+        job_categories = categories_from_job(job)
+        if not os.path.exists(path_dataset):
+            with open(ensure_dir(path_dataset), "w") as handler:
+                for asset in assets:
+                    response = requests.get(
+                        asset["content"],
+                        headers={
+                            "Authorization": f"X-API-Key: {self.api_key}",
+                        },
+                    )
+                    if job_name in asset["labels"][0]["jsonResponse"]:
+                        label_category = asset["labels"][0]["jsonResponse"][job_name]["categories"][
+                            0
+                        ]["name"]
+                        handler.write(
+                            json.dumps(
+                                {
+                                    "text": response.text,
+                                    "label": job_categories.index(label_category),
+                                }
+                            )
+                            + "\n"
+                        )
+                    else:
+                        asset_id = asset["id"]
+                        warn(f"Asset {asset_id} does not have {job_name} annotation")
+        raw_datasets = datasets.load_dataset(
+            "json",
+            data_files=path_dataset,
+            features=datasets.features.features.Features(
+                {
+                    "label": datasets.ClassLabel(names=job_categories),
+                    "text": datasets.Value(dtype="string"),
+                }
+            ),
+        )
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+
+        def tokenize_function(examples):
+            return tokenizer(examples["text"], padding="max_length", truncation=True)
+
+        tokenized_datasets = raw_datasets.map(tokenize_function, batched=True)  # type: ignore
+        train_dataset = tokenized_datasets["train"]
+        path_model = os.path.join(
+            path, "model", model_framework, datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        )
+        if model_framework == ModelFramework.PyTorch:
+
+            model = AutoModelForSequenceClassification.from_pretrained(
+                model_name, num_labels=len(job_categories), id2label=dict(enumerate(job_categories))
+            )
+        if model_framework == ModelFramework.Tensorflow:
```

Comment:
this should be an elif. Missing else with raise exception error. Also the model and the tokenizer should be instantiated next to each other, in a dedicated function

Reply:
factorized the `(token, model)` logic